examples/linux-offsets.c

```diff
@@ -0,0 +1,240 @@
+{
+    fprintf(stderr, "Usage:\n");
+    fprintf(stderr, "    %s [OPTIONS...]\n", arg0);
+    fprintf(stderr, "\n");
+    fprintf(stderr, "Required one of:\n");
+    fprintf(stderr, "    -n, --name           Domain name\n");
+    fprintf(stderr, "    -d, --domid          Domain ID\n");
+    fprintf(stderr, "Required input:\n");
+    fprintf(stderr, "    -r, --json-kernel    The OS kernel's json profile\n");
+    fprintf(stderr, "Optional input:\n");
+    fprintf(stderr, "    -k, --only-kpgd      Only print KPGD value\n");
+    fprintf(stderr, "\n");
+    fprintf(stderr, "Example:\n");
+    fprintf(stderr, "    %s -n lin7vm -r /opt/kernel.json\n", arg0);
+    fprintf(stderr, "    %s --domid 17 --json-kernel /opt/kernel.json --only-kpgd\n", arg0);
+}
+
+int main(int argc, char **argv)
+{
+    vmi_mode_t mode;
+    int rc = 1;
+
+    void *domain = NULL;
+    uint64_t domid = VMI_INVALID_DOMID;
+    uint64_t init_flags = 0;
+    int only_output_kpgd = 0;
+
+    char *kernel_profile = NULL;
+    int long_index = 0;
+    char c;
+
+    const struct option long_opts[] = {
+        {"name", required_argument, NULL, 'n'},
+        {"domid", required_argument, NULL, 'd'},
+        {"json-kernel", required_argument, NULL, 'r'},
+        {"only-kpgd", no_argument, NULL, 'k'},
+    };
+
+    while ((c = getopt_long (argc, argv, "n:d:kr:", long_opts, &long_index)) != -1)
+        switch (c) {
+            case 'n':
+                domain = (void *)optarg;
+                init_flags |= VMI_INIT_DOMAINNAME;
+                break;
+            case 'd':
+                domid = strtoull(optarg, NULL, 0);
+                domain = (void *)&domid;
+                init_flags |= VMI_INIT_DOMAINID;
+                break;
+            case 'k':
+                only_output_kpgd = 1;
+                break;
+            case 'r':
+                kernel_profile = optarg;
+                break;
+            default:
+                show_usage(argv[0]);
+                return 1;
+        }
+
+    if (optind != argc) {
+        fprintf(stderr, "Unrecognized argument: %s\n", argv[optind]);
+        show_usage(argv[0]);
+        return 1;
+    }
+
+    if (!domain) {
+        fprintf(stderr, "You have to specify --name or --domid!\n");
+        show_usage(argv[0]);
+        return 1;
+    }
+
+    if ((init_flags & VMI_INIT_DOMAINNAME) && (init_flags & VMI_INIT_DOMAINID)) {
+        fprintf(stderr, "Both domain ID and domain name provided!\n");
+        show_usage(argv[0]);
+        return 1;
+    }
+
+    if (!kernel_profile) {
+        fprintf(stderr, "You have to specify path to kernel JSON profile!\n");
+        show_usage(argv[0]);
+        return 1;
+    }
+
+    if (VMI_FAILURE == vmi_get_access_mode(vmi, domain, init_flags, NULL, &mode)) {
+        printf("Failed to get access mode\n");
+        goto done;
+    }
+
+    /* initialize the libvmi library */
+    if (VMI_FAILURE == vmi_init(&vmi, mode, domain, init_flags | VMI_INIT_EVENTS, NULL, NULL)) {
```

Comment:
Doesn't look like `VMI_INIT_EVENTS` is necessary here, you aren't using any events.